3 - FightClub/Code/Objects/Boxer.cs

```diff
@@ -0,0 +1,94 @@
+﻿using ObjectOrientedProblems.Code.Enums;
+using ObjectOrientedProblems.Code.Interfaces;
+
+namespace ObjectOrientedProblems.Code
+{
+    internal class Boxer : IFighter
+    {
+        public Boxer()
+        {
+            Damage = 1;
+            Health = 10;
+            State = FighterState.Healthy;
+        }
+        public int Health { get; set; }
+
+        public int Damage { get; set; }
+
+        public FighterState State { get; set; }
+
+        public void DrinkPotion(IPotion potion)
+        {
+            if(Health != 0) { Health += potion.Healing; }
+
+            if(Health >10) { Health = 10; }
+
+            if (Health >= 10) { State = FighterState.Healthy; }
```

Comment:
These should be combined, your third if statement *CAN'T* be above 10, your previous logic makes it impossible.